File: working/project1/folders/rFolder/folder2/sum.R

# CANNR TM tool example showing R function to expose as a Web service.
# Copyright 2020 Pat Tendick ptendick@gmail.com
# All rights reserved
# Maintainer Pat Tendick ptendick@gmail.com

# Sum the values of the variable x in a data frame.
sumX <- function(df) {
  
  return(sum(df$x))
  
}

# Sample JSON input is
# [{"x":1},{"x":2},{"x":3}]

File: working/project1/folders/pyFolder/folder1/sum.py
"""
CANNR TM tool example showing Python function to expose as a Web service.
Copyright 2020 Pat Tendick ptendick@gmail.com
All rights reserved
Maintainer Pat Tendick ptendick@gmail.com
"""

import json
import numpy as np

# Module info
rtModuleInfo = {
    "moduleName": "Python cannR sum test module",
    "moduleVersion": "0.1.0"
    }

# Calculates the sum of the elements in an array.
# Sample input is {"data": [1,2,3]}
def calcSum(
    pInput     # Input dictionary
    ):

    data = pInput.get("data", None)
    if data:
        return {'sum': int(np.sum(np.array(data)))}
    
    return 0.0

